stor/dx.py

```diff
@@ -385,129 +1067,161 @@ def exists(self):
+        """
+        if not self.resource:
+            return dxpy.DXProject(dxid=self.canonical_project).describe()
+        return dxpy.DXFile(dxid=self.canonical_resource,
+                           project=self.canonical_project).describe()
 
 
 class DXVirtualPath(DXPath):
-    """Class Handler for DXPath of form 'dx://project-{ID}:/a/b/c' or 'dx://a/b/c'"""
+    """Class Handler for DXPath of form 'dx://project-{dxID}:/a/b/c' or 'dx://a:/b/c'"""
 
     @property
     def virtual_project(self):
+        """Returns the virtual name of the project associated with the DXVirtualPath"""
         if utils.is_valid_dxid(self.project, 'project'):
-            return dxpy.DXProject(dxid=self.project).name
+            with _propagate_dx_exceptions():
+                return dxpy.DXProject(dxid=self.project).name
         return self.project
 
     @property
     def virtual_resource(self):
+        """Returns the human readable path of the resource associated with the DXVirtualPath"""
         return self.resource
 
     @property
     def virtual_path(self):
+        """Returns the DXVirtualPath instance"""
         return self
 
     @cached_property
     def canonical_project(self):
-        """Returns the unique project that matches the name that user has view access to.
-        If no match is found, returns None
+        """Returns the ID of the unique project that matches the name that user has at least
+        view access to. If no match is found, returns None
 
         Raises:
-            DuplicateProjectError - if project name is not unique on DX platform
-            NotFoundError - If project name doesn't exist on DNAnexus
+            DuplicateProjectError: If project name is not unique on DX platform
+            NotFoundError: If project name doesn't exist on DNAnexus
         """
         if utils.is_valid_dxid(self.project, 'project'):
             return self.project
-        else:
+
+        with _propagate_dx_exceptions():
             try:
                 proj_dict = dxpy.find_one_project(
                     name=self.project, level='VIEW', zero_ok=True, more_ok=False)
-                if proj_dict is None:
-                    raise ProjectNotFoundError('No projects were found with given name ({})'
-                                               .format(self.project))
-                return proj_dict['id']
-            except DXError as e:
+            except DXSearchError as e:
                 raise DuplicateProjectError('Duplicate projects were found with given name ({})'
                                             .format(self.project), e)
 
+        if proj_dict is None:
+            raise ProjectNotFoundError('No projects were found with given name ({})'
+                                       .format(self.project))
+
+        return proj_dict['id']
+
     @cached_property
     def canonical_resource(self):
         """Returns the dx file-ID of the uniquely matched filename
```

Comment:
```suggestion
        """The dxid of the file at this path
```

It's a property so doesn't return anything